extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/JandexUtil.java

```diff
@@ -201,4 +206,40 @@ public static boolean containsMethod(ClassInfo classInfo, MethodInfo methodInfo)
         }
         return false;
     }
+
+    public static ClassInfo getEnclosingClass(AnnotationInstance annotationInstance) {
+        switch (annotationInstance.target().kind()) {
+            case FIELD:
+                return annotationInstance.target().asField().declaringClass();
+            case METHOD:
+                return annotationInstance.target().asMethod().declaringClass();
+            case METHOD_PARAMETER:
+                return annotationInstance.target().asMethodParameter().method().declaringClass();
+            case CLASS:
+                return annotationInstance.target().asClass();
+            case TYPE:
+                return annotationInstance.target().asType().asClass(); // TODO is it legal here or should I throw ?
+            default:
+                throw new RuntimeException(); // this should not occurs
+        }
+    }
+
+    public static boolean isChildOf(IndexView index, ClassInfo info, DotName parentName) throws BuildException {
```

Comment:
```suggestion
    public static boolean isSubclassOf(IndexView index, ClassInfo info, DotName parentName) throws BuildException {
```

Given that this is only for superclasses (not interfaces) the name should make it clear.